packages/x-date-pickers-pro/src/DateTimeRangePicker/shared.tsx

```diff
@@ -0,0 +1,177 @@
+  /**
+   * Custom component for the toolbar rendered above the views.
+   * @default DateTimeRangePickerToolbar
+   */
+  Toolbar?: React.JSXElementConstructor<DateTimeRangePickerToolbarProps<TDate>>;
+}
+
+export interface BaseDateTimeRangePickerSlotsComponentsProps<TDate>
+  extends DateRangeCalendarSlotsComponentsProps<TDate> {
+  /**
+   * Props passed down to the tabs component.
+   */
+  tabs?: ExportedDateTimeRangePickerTabsProps;
+  /**
+   * Props passed down to the toolbar component.
+   */
+  toolbar?: ExportedDateTimeRangePickerToolbarProps;
+}
+
+export interface BaseDateTimeRangePickerProps<TDate>
+  extends Omit<
+      BasePickerInputProps<
+        DateRange<TDate>,
+        TDate,
+        DateTimeRangePickerViews,
+        DateRangeValidationError
+      >,
+      'orientation'
+    >,
+    ExportedDateRangeCalendarProps<TDate>,
+    BaseDateValidationProps<TDate>,
+    DesktopOnlyTimePickerProps<TDate> {
+  /**
+   * Overridable components.
+   * @default {}
+   * @deprecated Please use `slots`.
+   */
+  components?: BaseDateTimeRangePickerSlotsComponent<TDate>;
+  /**
+   * The props used for each component slot.
+   * @default {}
+   * @deprecated Please use `slotProps`.
+   */
+  componentsProps?: BaseDateTimeRangePickerSlotsComponentsProps<TDate>;
+  /**
+   * Overridable component slots.
+   * @default {}
+   */
+  slots?: UncapitalizeObjectKeys<BaseDateTimeRangePickerSlotsComponent<TDate>>;
+  /**
+   * The props used for each component slot.
+   * @default {}
+   */
+  slotProps?: BaseDateTimeRangePickerSlotsComponentsProps<TDate>;
+  /**
+   * Define custom view renderers for each section.
+   * If `null`, the section will only have field editing.
+   * If `undefined`, internally defined view will be the used.
+   */
+  viewRenderers?: Partial<
+    PickerViewRendererLookup<
+      DateRange<TDate>,
+      DateTimeRangePickerViews,
+      DateRangeViewRendererProps<TDate, 'day'>,
+      {}
+    >
+  >;
+}
+
+type UseDateTimeRangePickerDefaultizedProps<
+  TDate,
+  Props extends BaseDateTimeRangePickerProps<TDate>,
+> = LocalizedComponent<
+  TDate,
+  DefaultizedProps<Props, 'views' | 'openTo' | 'ampm' | keyof BaseDateValidationProps<TDate>>
+>;
+
+export function useDateTimeRangePickerDefaultizedProps<
+  TDate,
+  Props extends BaseDateTimeRangePickerProps<TDate>,
+>(
+  props: Props,
+  name: string,
+): UseDateTimeRangePickerDefaultizedProps<TDate, Omit<Props, 'components' | 'componentsProps'>> {
+  const utils = useUtils<TDate>();
+  const defaultDates = useDefaultDates<TDate>();
+  const { components, componentsProps, ...themeProps } = useThemeProps({
+    props,
+    name,
+  });
+
+  const localeText = React.useMemo<PickersInputLocaleText<TDate> | undefined>(() => {
+    if (themeProps.localeText?.toolbarTitle == null) {
+      return themeProps.localeText;
+    }
+
+    return {
+      ...themeProps.localeText,
+      dateRangePickerToolbarTitle: themeProps.localeText.toolbarTitle,
```

Comment:
Good observation. 🤔

Reply:
So you don't think we should use `dateTimeRangePickerToolbarTitle` instead?

Reply:
Sorry for not fixing it sooner.
This part is currently irrelevant on the Date Time Range Picker because the toolbar uses `start` and `end` translations for headers.
https://github.com/mui/mui-x/pull/9528/commits/423c2aecb40012004f17b623bc25175fe97430bc